services/intersection-api/api/src/main/java/us/dot/its/jpo/ode/api/accessors/assessments/ConnectionOfTravelAssessment/ConnectionOfTravelAssessmentRepositoryImpl.java

```diff
@@ -1,74 +1,104 @@
 import org.springframework.data.mongodb.core.MongoTemplate;
 import org.springframework.data.mongodb.core.query.Criteria;
 import org.springframework.data.mongodb.core.query.Query;
 import org.springframework.stereotype.Component;
 import us.dot.its.jpo.conflictmonitor.monitor.models.assessments.ConnectionOfTravelAssessment;
-import us.dot.its.jpo.ode.api.ConflictMonitorApiProperties;
+import us.dot.its.jpo.ode.api.accessors.IntersectionCriteria;
+import us.dot.its.jpo.ode.api.accessors.PageableQuery;
+
+import java.util.Collections;
+import java.util.List;
 
 @Component
-public class ConnectionOfTravelAssessmentRepositoryImpl implements ConnectionOfTravelAssessmentRepository {
+public class ConnectionOfTravelAssessmentRepositoryImpl
+        implements ConnectionOfTravelAssessmentRepository, PageableQuery {
 
     private final MongoTemplate mongoTemplate;
-    private final ConflictMonitorApiProperties props;
 
     private final String collectionName = "CmConnectionOfTravelAssessment";
+    private final String DATE_FIELD = "assessmentGeneratedAt";
+    private final String INTERSECTION_ID_FIELD = "intersectionID";
 
     @Autowired
-    public ConnectionOfTravelAssessmentRepositoryImpl(MongoTemplate mongoTemplate, ConflictMonitorApiProperties props) {
+    public ConnectionOfTravelAssessmentRepositoryImpl(MongoTemplate mongoTemplate) {
         this.mongoTemplate = mongoTemplate;
-        this.props = props;
-    }
-
-    public Query getQuery(Integer intersectionID, Long startTime, Long endTime, boolean latest) {
-        Query query = new Query();
-
-        if (intersectionID != null) {
-            query.addCriteria(Criteria.where("intersectionID").is(intersectionID));
-        }
-
-        Date startTimeDate = new Date(0);
-        Date endTimeDate = new Date();
-
-        if (startTime != null) {
-            startTimeDate = new Date(startTime);
-        }
-        if (endTime != null) {
-            endTimeDate = new Date(endTime);
-        }
-
-        query.addCriteria(Criteria.where("assessmentGeneratedAt").gte(startTimeDate).lte(endTimeDate));
-
-        if (latest) {
-            query.with(Sort.by(Sort.Direction.DESC, "assessmentGeneratedAt"));
-            query.limit(1);
-        } else {
-            query.limit(props.getMaximumResponseSize());
-        }
-        return query;
     }
 
-    public long getQueryResultCount(Query query) {
-        return mongoTemplate.count(query, ConnectionOfTravelAssessment.class, collectionName);
+    /**
+     * Get a page representing the count of data for a given intersectionID,
+     * startTime, and endTime
+     *
+     * @param intersectionID the intersection ID to query by, if null will not be
+     *                       applied
+     * @param startTime      the start time to query by, if null will not be applied
+     * @param endTime        the end time to query by, if null will not be applied
+     * @param pageable       the pageable object to use for pagination
```

Comment:
chore(blocking): remove unused param from this (and all other) `count` method java docs
```suggestion
     * @param endTime        the end time to query by, if null will not be applied
```

Reply:
Applied to all accessors